task5/university_members.py

```diff
@@ -0,0 +1,336 @@
+                    member = Professor(value["id"],
+                                    value["name"],
+                                    value["age"],
+                                    value["department"],
+                                    value["title"],
+                                    value["salary"])
+                    members.append(member)
+                elif member["type"] == "adminstaff":
+                    member = AdminStaff(value["id"],
+                                    value["name"],
+                                    value["age"],
+                                    value["position"],
+                                    value["years_of_service"],
+                                    value["salary"])
+                    members.append(member)
+                else:
+                    print(f"Unsupported type {member['type']} with value {member['value']}")
+    except KeyError as err:
+        pass
+    return members
+
+
+def deserialize_pkl(file_path: str) -> list:
+    """Deserializes a list of members from a pkl file."""
+
+    members = []
+
+    try:
+        if os.path.exists(file_path):   
+            with open(file_path, 'rb') as f:
+                members = pickle.load(f)
+    except KeyError as err:
+        pass
+    return members
+
+
+def show_members(members: list) -> None:
+    """Displays information about members on the screen."""
+
+    if not members:
+        print(f"\n{RED}There are no users!{RESET}")
+        return
+
+    for index, member in enumerate(members):
+        if isinstance(member, Student):
+            print(
+                f"\n{index}| student | id: {member.id}, name: {member.name}, "
+                f"age: {member.age}, course: {member.course}, "
+                f"year: {member.year}, gpa: {member.gpa}")
+        elif isinstance(member, Professor):
+            print(
+                f"\n{index}| professor | id: {member.id}, name: {member.name}, "
+                f"age: {member.age}, department: {member.department}, "
+                f"title: {member.title}, salary: {member.salary}")
+        elif isinstance(member, AdminStaff):
+            print(
+                f"\n{index}| admin_staff | id: {member.id}, name: {member.name}, "
+                f"age: {member.age}, position: {member.position}, "
+                f"years_of_service: {member.years_of_service}, salary: {member.salary}")
+
+
+def get_unused_id(members: list) -> int:
+    """Generates the next unique ID for the new member."""
+
+    existing_ids = {member.id for member in members}
+    next_id = 1
+    while next_id in existing_ids:
+        next_id += 1
+    return next_id
+
+
+def remove_member_by_id(members: list) -> None:
+    """Deletes a member by the given ID."""
+
+    try:
+        id_to_remove = int(input("\nEnter ID to remove: "))
+        for member in members:
+            if member.id == id_to_remove:
+                members.remove(member)
```

Comment:
This might look simpler:
```suggestion
                del member
```

Reply:
done